google-cloud-env/lib/google/cloud/env.rb

```diff
@@ -65,22 +65,55 @@ class Env
       # @private URL path for metadata server root.
       METADATA_ROOT_PATH = "/".freeze
 
+      # @private
+      METADATA_RETRY_EXCEPTIONS =
+        ::Faraday::Request::Retry::DEFAULT_EXCEPTIONS +
+        [::Errno::ETIMEDOUT, ::Faraday::ConnectionFailed, ::Errno::EHOSTDOWN]
+
       ##
       # Create a new instance of the environment information.
       # Most client should not need to call this directly. Obtain a singleton
       # instance of the information from `Google::Cloud.env`. This constructor
       # is provided for internal testing and allows mocking of the data.
       #
       # @param [Hash] env Mock environment variables.
+      # @param [Hash,false] metadata_cache The metadata cache. You may pass
+      #     a prepopuated cache, an empty cache (the default) or `false` to
+      #     disable the cache completely.
+      # @param [Numeric] request_timeout Timeout for each http request.
+      #     Defaults to 0.1.
+      # @param [Integer] retries Number of times to retry http requests.
+      #     Defaults to 0.
+      # @param [Proc] build_proc A proc to be used to build the Faraday
+      #     connection. It should take a `Faraday::RackBuilder` as its single
+      #     parameter, and should add middleware and set the adapter as if it
+      #     were passed as a block to `Faraday.new`.
       # @param [Faraday::Connection] connection Faraday connection to use.
-      # @param [Hash] metadata_cache Mock cache.
+      #     If specified, overrides the `request_timeout`, `retries`, and
+      #     `build_proc` settings.
       #
-      def initialize env: nil, connection: nil, metadata_cache: nil
+      def initialize env: nil, connection: nil, metadata_cache: nil,
+                     request_timeout: 0.1, retries: 0, build_proc: nil
```

Comment:
How will these defaults be set on the `Google::Cloud.env` instance?

Reply:
Yeah, good question. They're currently set to match the previous behavior (100ms timeout, and no retry) but perhaps it should default to something else. I'll try setting it to one retry. (We have an internal App Engine prober that seems to be failing intermittently on metadata server queries, so we'll have a real world test case to measure the effect.)